leechcore_device_microvmi/leechcore_device_microvmi.c

```diff
@@ -24,48 +24,84 @@ static VOID DeviceMicrovmi_Close(_Inout_ PLC_CONTEXT ctxLC)
+        char* param_value = strtok_r(NULL, "=", &saveptr2);
+        if (!strncmp(param_name, "vm_name", strlen("vm_name"))) {
+            init_params->common.vm_name = strdup(param_value);
+        } else if (!strncmp(param_name, "kvm_unix_socket", strlen("kvm_unix_socket"))) {
+            init_params->kvm.tag = UnixSocket;
+            init_params->kvm.unix_socket.path = strdup(param_value);
+        } else {
+          lcprintfv(ctxLC, "MICROVMI: unhandled init parameter: %s\n", param_name);
+        }
+    }
+    free(szDevice_start_params);
+    return true;
+}
+
 _Success_(return)
 EXPORTED_FUNCTION BOOL LcPluginCreate(_Inout_ PLC_CONTEXT ctxLC, _Out_opt_ PPLC_CONFIG_ERRORINFO ppLcCreateErrorInfo)
 {
-  lcprintfv(ctxLC, "MICROVMI: Initializing\n");
-  // safety checks
-	if(ppLcCreateErrorInfo) { *ppLcCreateErrorInfo = NULL; }
-	if(ctxLC->version != LC_CONTEXT_VERSION) { return false; }
-  
-  // TODO: handle init args
-  // check URL scheme
-  if (strncmp(ctxLC->Config.szDevice, PLUGIN_URL_SCHEME, strlen(PLUGIN_URL_SCHEME))) {
-    // no match, quit
-    return false;
-  }
+    lcprintfv(ctxLC, "MICROVMI: Initializing\n");
+    // safety checks
+    if(ppLcCreateErrorInfo) { *ppLcCreateErrorInfo = NULL; }
+    if(ctxLC->version != LC_CONTEXT_VERSION) { return false; }
+    
+    // handle init args
+    DriverInitParamsFFI init_params = {0};
+    if (!parse_url_args(ctxLC, &init_params)) {
+      return false;
+    }
 
-  // TODO: init libmicrovmi
-  microvmi_envlogger_init();
-  const char* init_error = NULL;
-  DriverInitParamsFFI init_params = {
-    .common = {
-      .vm_name = "win10"
-    },
-    .kvm = {
-      .tag = UnixSocket,
-      .unix_socket.path = "/tmp/introspector",
+    // init libmicrovmi
+    microvmi_envlogger_init();
+    const char* init_error = NULL;
+    
+    void* microvmi_driver = microvmi_init(NULL, &init_params, &init_error);
+    if (!microvmi_driver) {
+      lcprintfv(ctxLC, "MICROVMI: initialization failed: %s.\n", init_error);
+      goto error_exit;
     }
-  };
-  void* microvmi_driver = microvmi_init(NULL, &init_params, &init_error);
-  if (!microvmi_driver) {
-    lcprintf(ctxLC, "Device Microvmi initialization failed: %s.\n", init_error);
-    rs_cstring_free((char*)init_error);
-    return false;
-  }
 
-  // assign context
-  ctxLC->hDevice = (HANDLE)microvmi_driver;
+    // assign context
+    ctxLC->hDevice = (HANDLE)microvmi_driver;
 
-  // setup config
-  ctxLC->Config.fVolatile = true;
-  // set callback functions
-  ctxLC->pfnReadContigious = DeviceMicrovmi_ReadContigious;
-  ctxLC->pfnClose = DeviceMicrovmi_Close;
-  // status
-  lcprintfv(ctxLC, "MICROVMI: initialized.\n");
+    // setup config
+    ctxLC->Config.fVolatile = true;
+    // set callback functions
+    ctxLC->pfnReadContigious = DeviceMicrovmi_ReadContigious;
+    ctxLC->pfnClose = DeviceMicrovmi_Close;
+    // status
+    lcprintfv(ctxLC, "MICROVMI: initialized.\n");
+    return true;
```

Comment:
I notice that you fixed some issues from the previous commit in this one. You should squash unnecessary intermediate commits, so that each commit is correct and stands on its own.